### DISCUSSION TITLE
Vercel Serverless Function Memory Limit

### DISCUSSION THREAD

**Discussion by yoga-teachedison** [vercel/next.js#40248] [Help]
I have a Next.js project deployed in Vercel. I recently introduced Fabric.js to my project, and the build started failing because of Memory Issue on Vercel, whereas it is successful on my local machine.

All my pages are serverless functions as they contain getInitialProps. There was no problem until I introduced Fabric.js, which includes CANVAS(~40MB), and Vercel has a memory limit for any serverless functions of 50MB, which can't go beyond that. Because of this, the builds are failing on Vercel. 

<img width="965" alt="Screenshot 2022-09-05 at 7 21 26 PM" src="https://user-images.githubusercontent.com/75308636/188464975-824ecc80-4db9-4eb1-a471-77089e6eac62.png">

Surprisingly, pages that don't import Fabric.js also fail because of memory issues. In the above screenshot, the explorer page build is failing because of Fabric.js, but it is not even imported/referred to on that page. 

I have two questions here,
1.) How do you think we could solve this issue without compromising the serverless functions? i.e. I don't want to remove getInitialProps.
2.) I can understand the Page having a dependency with Fabric.js failing, but why does a page that doesn't have a dependency with that package fail while building on Vercel?

Looking for your valuable thoughts

**Comment by jeetiss** [comment]
Hello @yoga-teachedison
use [@napi-rs/canvas](https://github.com/Brooooooklyn/canvas) package. it supports vercel out of the box and doesn't have any problems with Memory Limits (package size `~9MB`)
